### DISCUSSION TITLE
hombrew/homebrew-test-bot reports errors for ci run with no details

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#3234] [Tap maintenance and Homebrew development] [answered]
attempting to build some bottles for the homebrew/homebrew-freecad tap using homebrew-test-bot but unfortunately viewing the logs for the ci runs via the actions tab on the github web interface is showing errors. is there a way where i can get more details on why the tests are failing?

https://github.com/FreeCAD/homebrew-freecad/runs/6279174121?check_suite_focus=true#step:17:1685

**Comment by Bo98** [comment, marked as answer]
```
   ==> /usr/local/Cellar/python@3.10/3.10.2/bin/python3 -m ensurepip
  Last 150 lines from /Users/brewer/Library/Logs/Homebrew/python@3.10/post_install.01.python3:
  2022-05-03 14:26:01 -0500
  
  /usr/local/Cellar/python@3.10/3.10.2/bin/python3
  -m
  ensurepip
  
  dyld: Library not loaded: /usr/local/opt/gettext/lib/libintl.8.dylib
    Referenced from: /usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/Python
    Reason: image not found
  You can try again using:
    brew postinstall python@3.10
  Warning: The post-install step did not complete successfully
```

Looks like you have some custom version of Python? The bottles supplied by us are for 3.10.4 and do not link with `gettext`.

**Reply by ipatch** [reply]
thanks. i'll do some investigation on that. FWIW those steps showed up as warnings not errors 🤔

**Reply by ipatch** [reply]
so if my formula are using python@3.10 but python@3.10 is `keg_only` what's the way to make `test-bot` happy when attempting to python@3.10 ?

**Reply by ipatch** [reply]
logging into the self-hosted runner, and manually running those commands above i don't experience such error

```
brew postinstall python@3.10
==> Postinstalling python@3.10
==> /usr/local/Cellar/python@3.10/3.10.2/bin/python3 -m ensurepip
==> /usr/local/Cellar/python@3.10/3.10.2/bin/python3 -m pip install -v --no-deps --no-index --upgrade --isolated --target=/usr/local/lib/python3.10/sit
vmmojave:~ brewer$
```

**Reply by carlocab** [reply]
Looks like your Mojave bottle has unexpected linkage to `gettext`. CC @gromgit

**Reply by Bo98** [reply]
> logging into the self-hosted runner, and manually running those commands above i don't experience such error

It won't if you happened to have gettext already installed. `brew test-bot` takes a clean slate, as it's supposed to. You're seeing a miscompilation with the Python bottle you are using.

**Reply by gromgit** [reply]
Yeah, sorry, `gettext` seems to have been picked up as an opportunistic linkage.

@ipatch, I've just rebuilt the latest `python@3.10`, and confirmed that it no longer links in `gettext`, please see if it works for you.

**Reply by ipatch** [reply]
i saw that. thank you 🙏
